### PR TITLE
burn_vector_geometry and polygonize

### PR DESCRIPTION
**Pull request by Huite**
Add vector utilities such as provided by GDAL for raster data.

### CODE REVIEW

File: xugrid/ugrid/burn.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    # Find the associated values.
+    line_index = index[1:][valid]
+    value_index = line_index[edge_index]
+    output[face_index] = values[value_index]
+    return
+
+
+def burn_vector_geometry(
+    gdf: "geopandas.GeoDataframe",  # type: ignore # noqa
+    like: "xugrid.Ugrid2d",
```
Comment (JoerivanEngelen) [inline xugrid/ugrid/burn.py:226]: I was a bit in doubt about the name ``like`` here, conceptually I perceived this method different than ``regrid_like`` or ``full_like``. As it is returning a copy of ``like`` however, it is essentially doing such a thing, but it feels a bit unintuitive. I cannot think of a better name though. ``grid`` doesn't cover it (as UgridDataset is also allowed), and ``template`` introduces suddenly new nomenclature to the package.  Just sharing my thoughts, you can resolve without notice.

**Review by JoerivanEngelen — COMMENTED** [review]
Looks good, my comments mostly concern comments, documentation and type annotation.
I got one comment to provide an example to burn points into a grid which would require a bit of work though.

File: examples/vector_conversion.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    ax.set_ylim(ymin, ymax)
+    ax.set_title(title)
+
+# %%
+# This function can also be used to burn points or lines into the faces of an
+# unstructured grid.
+#
+# The exterior boundaries of the province polygons will provide
+# a collection of linestrings that we can burn into the grid:
+
```
[the remark this answers is not in the document]
Comment (Huite) [reply examples/vector_conversion.py:137]: Done using the centroids of the provinces.

File: examples/vector_conversion.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+#
+# Rasterizing is a common operation when working with raster and vector data.
+# While we cannot name the operation "rasterizing" when we're dealing with
+# unstructured grids, there is a clearly equivalent operation where we mark
+# cells that are covered or touched by a polygon.
+#
+# In this example, we mark the faces that are covered by a certain province.
+
+provinces = xu.data.provinces_nl().to_crs(28992)
+provinces["value"] = range(len(provinces))
```
[the remark this answers is not in the document]
Comment (Huite) [reply examples/vector_conversion.py:78]: Added to the extra paragraph.

File: xugrid/ugrid/burn.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+
+@nb.njit(inline="always", parallel=True, cache=True)
+def points_in_triangles(
+    points: FloatArray,
+    face_indices: IntArray,
+    faces: IntArray,
+    vertices: FloatArray,
+):
+    # TODO: move this into numba_celltree instead?
```
[the remark this answers is not in the document]
Comment (Huite) [reply xugrid/ugrid/burn.py:93]: Done

**Pull request merged**